unit_tests/testFileType_XDMF/testFileType_XDMF.f90

```diff
@@ -0,0 +1,618 @@
+    1.2720399211630999,       0.45277135515826000,        0.0000000000000000,&
+    1.2720399211630999,        1.5472286448417001,        0.0000000000000000,&
+    1.4199853588381000,        1.6778602930840001,        0.0000000000000000,&
+    1.4199853588381000,       0.32213970691595001,        0.0000000000000000,&
+    1.4596903386044000,        1.5215473894763001,        0.0000000000000000,&
+    1.4596903386044000,       0.47845261052367000,        0.0000000000000000 &
+/), (/3, 109/))
+
+INTEGER(SLK) :: two_pins_pin1_cells(7,46) = RESHAPE( (/ &
+36,  33,  46,   4,  47,  48,  39,&
+36,  34,  46,  33,  49,  47,  40,&
+36,   4,  46,  38,  48,  50,  45,&
+36,  35,  46,  34,  51,  49,  41,&
+36,  36,  46,  35,  52,  51,  42,&
+36,  38,  46,  37,  50,  53,  44,&
+36,  37,  46,  36,  53,  52,  43,&
+36,  13,  36,  35,  62,  42,  63,&
+36,  14,  36,  13,  64,  62,  17,&
+36,  12,  13,  35,  16,  63,  65,&
+36,   5,   6,  37,   9,  66,  67,&
+36,  28,  34,  27,  68,  69,  31,&
+36,  28,  54,  34,  70,  71,  68,&
+36,  37,  55,   5,  72,  73,  67,&
+36,  35,  54,  12,  74,  75,  65,&
+36,  14,  55,  36,  76,  77,  64,&
+36,  19,  59,   1,  78,  79,  22,&
+36,   3,  58,  21,  80,  81,  25,&
+36,   5,  55,   0,  73,  82,   8,&
+36,   0,  55,  14,  82,  76,  18,&
+36,  12,  54,   2,  75,  83,  15,&
+36,   2,  54,  28,  83,  70,  32,&
+36,  34,  54,  35,  71,  74,  41,&
+36,  36,  55,  37,  77,  72,  43,&
+36,   1,  59,   7,  79,  84,  11,&
+36,  26,  58,   3,  85,  80,  29,&
+36,   6,  61,  37,  86,  87,  66,&
+36,  34,  60,  27,  88,  89,  69,&
+36,   4,  56,  20,  90,  91,  92,&
+36,  20,  57,   4,  93,  94,  92,&
+36,  21,  57,  20,  95,  93,  24,&
+36,  20,  56,  19,  91,  96,  23,&
+36,   7,  61,   6,  97,  86,  10,&
+36,  27,  60,  26,  89,  98,  30,&
+36,  38,  56,   4,  99,  90,  45,&
+36,   4,  57,  33,  94, 100,  39,&
+36,  56,  59,  19, 101,  78,  96,&
+36,  21,  58,  57,  81, 102,  95,&
+36,  37,  61,  38,  87, 103,  44,&
+36,  33,  60,  34, 104,  88,  40,&
+36,  26,  60,  58,  98, 105,  85,&
+36,  59,  61,   7, 106,  97,  84,&
+36,  58,  60,  33, 105, 104, 107,&
+36,  38,  61,  59, 103, 106, 108,&
+36,  38,  59,  56, 108, 101,  99,&
+36,  57,  58,  33, 102, 107, 100 &
+/), (/7, 46/))
+
+INTEGER(SNK) :: two_pins_pin1_material_ids(46) = (/ &
+0, 0, 0, 0, 0, 0, 0,          &
+1, 1, 1, 1, 1, 1, 1, 1, 1, 1, &
+1, 1, 1, 1, 1, 1, 1, 1, 1, 1, &
+1, 1, 1, 1, 1, 1, 1, 1, 1, 1, &
+1, 1, 1, 1, 1, 1, 1, 1, 1     &
+/)
+
+REAL(SDK) :: three_level_grid_L3_vertices(3,5) = RESHAPE( (/ &
+ 2.0, 1.5, 0.0,&
+ 2.0, 1.0, 0.0,&
+ 3.0, 1.0, 0.0,&
+ 2.0, 2.0, 0.0,&
+ 3.0, 2.0, 0.0 &
+/), (/3, 5/))
+
+
+INTEGER(SLK) :: three_level_grid_L3_cells(3,3) = RESHAPE( (/ &
+0, 2, 4, &
+1, 2, 0, &
+3, 0, 4  &
+/), (/3, 3/))
+
+
+CREATE_TEST('XDMF TYPE')
+REGISTER_SUBTEST('two_pins',test_two_pins)
+REGISTER_SUBTEST('three_level_grid',test_three_level_grid)
+REGISTER_SUBTEST('three_level_grid_IH', test_three_level_grid_implicit_hierarchy)
```

Comment:
SUBTEST should be to test each procedure (constructor/destructor, type-bound procedure, etc.) rather than each type of problems. COMPONENT_TEST may be used to test each case/problem.